pkg/generate/code/set_resource.go

```diff
@@ -119,13 +119,46 @@ func SetResource(
 	}
 
 	// We might be in a "wrapper" shape. Unwrap it to find the real object
-	// representation for the CRD's createOp. If there is a single member
-	// shape and that member shape is a structure, unwrap it.
-	if outputShape.UsedAsOutput && len(outputShape.MemberRefs) == 1 {
-		for memberName, memberRef := range outputShape.MemberRefs {
-			if memberRef.Shape.Type == "structure" {
-				sourceVarName += "." + memberName
-				outputShape = memberRef.Shape
+	// representation for the CRD's createOp/DescribeOP.
+
+	// Use the wrapper field path if it's given in the ack-generate config file.
+	wrapperFieldPath := r.SetOutputWrapperFieldPath(op)
+	if wrapperFieldPath != nil {
+		wrapperFieldParts := strings.Split(*wrapperFieldPath, ".")
+	mainLoop:
```

Comment:
Interested to hear opinions about use of goto labels. I understand you want to continue the outer loop, but I imagine this could be achieved with booleans - although messier.

Reply:
I don't have a strong opinion about `goto` statements - I personally only use them sometimes for `continue`/`break` statements. I'll push a different approach using using booleans

Reply:
I don't believe that boolean logic here would be any cleaner. I think this is a fair use of `goto` statements.

Reply:
Indeed it will be a little bit messier :)
Thoughts @vijtrip2 @jaypipes ?

Reply:
How about breaking the below code out into a helper function?

Reply:
Done, and added unit tests for the helper function